### PR TITLE
Adds basic setup and rubocop

### LINKED ISSUE #1: Error while seeding db
### Steps to reproduce 

Run `rake db:seed` after `rake db:create`

### Error Log 
```sh
rake aborted!
ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "notebooks_pkey"
DETAIL:  Key (id)=(1) already exists.
: INSERT INTO "notebooks" ("id", "name", "desc", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"

....

Caused by:
PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "notebooks_pkey"
DETAIL:  Key (id)=(1) already exists.
```

### LINKED ISSUE #8: Add Rubocop
see https://github.com/rubocop-hq/rubocop

### LINKED ISSUE #11: Add unneccessary files and folders to .gitignore
I think the following files and folders need to be kept under `.gitignore`

- [x] .keep
- [x] logs/
- [x] bin/
- [x] tmp/

### PR DESCRIPTION
**Pull request by electron0zero**
**Why?**
Fixes the `rake db:seed` error and adds Rubocop for linting

**CHANGELOG** 
1. Updated README
2. Fixed `db:seed` and added basic setup instructions/methods
3. Added [rubocop](https://rubocop.readthedocs.io/en/latest/) for linting
4. Modified .gitignore

Fixes #1
Fixes #8 
Fixes #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+- Create `database.yml` from `database.yml.example` file and add your username and password
+
+### Database initialization
+- run `bundle exec rails db:create db:schema:load db:seed` to create db and add seed data
+- run `bundle exec rails db:drop db:create db:schema:load db:seed` to drop current db and start fresh
+
+### How to run the test suite
+We use [Rspec](http://rspec.info/) with [factory_bot](https://github.com/thoughtbot/factory_bot) for testing
+
+- run `bundle exec rspec` to run full test suite
```
Comment (rishacha) [inline README.md]: I executed this, but got the following. There are no examples yet 
```sh
No examples found.


Finished in 0.00039 seconds (files took 0.11055 seconds to load)
0 examples, 0 failures
```

File: config/database.yml.example [hunk 2]
```diff
@@ -28,7 +29,7 @@ development:
   database: nb_development
   pool: 5
   username: nb_user
-  password:
+  password: pass
```
Comment (rishacha) [inline config/database.yml.example]: Should we hardcode the `pass` here ? If it's fine with you, you can ignore this comment.

File: README.md [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
+- run `bundle exec rspec <test-file/folder>` to run parts of suite
+- add `-fd` flag to run test with formatted output
 
 * Services (job queues, cache servers, search engines, etc.)
```
Comment (rishacha) [inline README.md]: Now how do I run it ?

Comment (rishacha) [reply README.md]: I ran it using this  `rails server -p 9000`. It worked, but it's serving the default page 
image

We need a swagger ui implementation here. I will implement the swagger ui part

File: config/database.yml.example (hunk 2 above)
Comment (electron0zero) [reply config/database.yml.example]: It's example file, I left it here to make it easy to setup

File: README.md (hunk 1 above)
Comment (electron0zero) [reply README.md]: if you run it from places other the root folder of repo, you will get that output.

run it from root folder of repo.

```sh
electron@Inspiron-3542:~/wd/nb/noteblocks(init) $ ls
app  bin  config  config.ru  db  Gemfile  Gemfile.lock  lib  log  public  Rakefile  README.md  spec  storage  tmp  vendor
electron@Inspiron-3542:~/wd/nb/noteblocks(init) $ bundle exec rspec
....................

Finished in 0.4331 seconds (files took 0.68178 seconds to load)
20 examples, 0 failures

electron@Inspiron-3542:~/wd/nb/noteblocks(init) $
```

File: README.md (hunk 3 above)
Comment (electron0zero) [reply README.md]: if you want to play around, launch REPL with `rails console` or `rails c`
you already used`rails server`

running `rails dbconsole` will drop you into psql/db prompt 

run `rails routes` to see all possible routes to query(or you can see controller tests where I query some routes).

**Pull request merged**